lib/librte_regexdev/rte_regexdev_core.h

```diff
@@ -0,0 +1,168 @@
+				    	    struct rte_regex_dev_xstats_map
+					    *xstats_map);
+/**< @internal Get xstats name map for the regex device. */
+
+typedef int (*regex_dev_xstats_get_t)(struct rte_regex_dev *dev,
+				      const uint16_t ids[], uint64_t values[],
+				      uint16_t n);
+/**< @internal Get xstats values for the regex device. */
+
+typedef int (*regex_dev_xstats_by_name_get_t)(struct rte_regex_dev *dev,
+					      const char *name, uint64_t value);
+/**< @internal Get xstat value for the regex device based on the xstats name. */
+
+typedef int (*regex_dev_xstats_reset_t)(struct rte_regex_dev *dev,
+					const uint16_t ids[], uint64_t values[],
+					uint16_t n);
+/**< @internal Reset xstats values for the regex device. */
+
+typedef int (*regex_dev_xstats_reset_t)(struct rte_regex_dev *dev,
+					const uint16_t ids[], uint64_t values[],
+					uint16_t n);
+/**< @internal Reset xstats values for the regex device. */
+
+typedef int (*regex_dev_selftest_t)(struct rte_regex_dev *dev);
+/**< @internal Trigger regex self test. */
+
+typedef int (*regex_dev_dump_t)(struct rte_regex_dev *dev, FILE *f);
+/**< @internal Dump internal information about the regex device. */
+
+typedef int (*regex_dev_enqueue_t)(struct rte_regex_dev *dev, uint16_t qp_id,
+				   struct rte_regex_ops **ops, uint16_t nb_ops);
+/**< @internal Enqueue a burst of scan requests to a queue on regex device. */
+
+typedef int (*regex_dev_dequeue_t)(struct rte_regex_dev *dev, uint16_t qp_id,
+				   struct rte_regex_ops **ops,
+				   uint16_t nb_ops);
+/**< @internal Dequeue a burst of scan response from a queue on regex device. */
+
+/**
+ * regex device operations
+ */
+struct rte_regex_dev_ops {
+	regex_dev_info_get_t dev_info_get;
+	regex_dev_configure_t dev_configure;
+	regex_dev_qp_setup_t dev_qp_setup;
+	regex_dev_start_t dev_start;
+	regex_dev_stop_t dev_stop;
+	regex_dev_close_t dev_close;
+	regex_dev_rule_db_update_t dev_rule_db_update;
+	regex_dev_rule_db_import_t dev_db_import;
+	regex_dev_rule_db_export_t dev_db_export;
+	regex_dev_xstats_names_get_t dev_xstats_names_get;
+	regex_dev_xstats_get_t dev_xstats_get;
+	regex_dev_xstats_by_name_get_t dev_xstats_by_name_get;
+	regex_dev_xstats_reset_t dev_xstats_reset;
+	regex_dev_selftest_t dev_selftest;
+	regex_dev_dump_t dev_dump;
+
+	/** Reserved for future extension */
+	void *reserved[5];
+};
+
+/**
+ * @internal
+ * The data part, with no function pointers, associated with each regex device.
+ *
+ * This structure is safe to place in shared memory to be common among different
+ * processes in a multi-process configuration.
+ */
+struct rte_regex_dev_data {
+	char name[RTE_REGEX_NAME_MAX_LEN]; /**< Unique identifier name */
+	uint16_t nb_qp; /**< Number of queue pairs. */
+	void *dev_private; /**< PMD-specific private data. */
+	struct rte_regex_dev_config dev_conf;
+	/**< Configuration applied to device. */
+	uint16_t port_id;           /**< Device [external] port identifier. */
```

Comment:
we do not have concept of ports (which exist for eth/eth switch etc) for regex device.
So please rename it to more relevant one. device_id? device_index?

Reply:
will change to device_id